### PR TITLE
feat:Support of subscribing eventing in custom component

### PR DESCRIPTION
**Pull request by jalagari**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #<gh-issue-id>

Test URLs:
- Before: https://subscribe--aem-boilerplate-forms--adobe-rnd.hlx.live/

### CODE REVIEW

File: blocks/form/util.js [hunk 1]
```diff
@@ -233,3 +233,18 @@ export function checkValidation(fieldElement) {
...
   updateOrCreateInvalidMsg(fieldElement, message);
 }
+
+export function subscribe(fieldDiv, callback) {
+  if (callback) {
+    fieldDiv.dataset.subscribe = true;
+    const observer = new MutationObserver((mutationsList) => {
+      mutationsList?.forEach((mutation) => {
+        if (mutation.type === 'attributes' && mutation.attributeName === 'data-field-model') {
+          callback(fieldDiv, JSON.parse(fieldDiv.dataset.fieldModel));
```
Comment (vdua) [inline blocks/form/util.js:243]: this fieldModel will not suffice. We decided we will somehow pass the model object not the state.

File: blocks/form/rules/index.js [hunk 2]
```diff
@@ -182,6 +179,9 @@ async function fieldChanged(payload, form, generateFormRendition) {
         break;
     }
   });
+  if (fieldWrapper?.dataset?.subscribe) {
+    fieldWrapper.dataset.fieldModel = JSON.stringify(fieldModel);
```
Comment (vdua) [inline blocks/form/rules/index.js:183]: this will be too big for a panel

File: blocks/form/util.js (hunk 1 above)
Comment (vdua) [reply blocks/form/util.js:243]: I would suggest moving this to form/index.js where we can have access to form object and pass the form model in the callback

**Comment by nit23uec** [comment]
closing this, as it will be covered by https://github.com/adobe-rnd/aem-boilerplate-forms/pull/142

**Pull request closed without merging**